src/executor/query/GetVerticesExecutor.cpp

```diff
@@ -26,23 +27,28 @@ folly::Future<Status> GetVerticesExecutor::getVertices() {
     SCOPED_TIMER(&execTime_);
 
     auto *gv = asNode<GetVertices>(node());
-
     GraphStorageClient *storageClient = qctx()->getStorageClient();
     nebula::DataSet vertices({kVid});
-    const auto& spaceInfo = qctx()->rctx()->session()->space();
     if (gv->src() != nullptr) {
         // Accept Table such as | $a | $b | $c |... as input which one column indicate src
         auto valueIter = ectx_->getResult(gv->inputVar()).iter();
-        VLOG(1) << "GV input var: " << gv->inputVar() << " iter kind: " << valueIter->kind();
+        VLOG(3) << "GV input var: " << gv->inputVar() << " iter kind: " << valueIter->kind();
         auto expCtx = QueryExpressionContext(qctx()->ectx());
+        const auto &spaceInfo = qctx()->rctx()->session()->space();
+        std::unordered_set<Value> uniqueSet;
         for (; valueIter->valid(); valueIter->next()) {
             auto src = gv->src()->eval(expCtx(valueIter.get()));
-            VLOG(1) << "src vid: " << src;
             if (!SchemaUtil::isValidVid(src, spaceInfo.spaceDesc.vid_type)) {
                 LOG(WARNING) << "Mismatched vid type: " << src.type();
                 continue;
             }
-            vertices.emplace_back(Row({std::move(src)}));
+            if (gv->dedup()) {
```

Comment:
This dedup option is applied for the result not the input.But now we return vid always so it's workable. The default return column is plan to remove.

Reply:
Is there some problems? If we dedup the input vid of vertices, GV always returns unique result set whether we pass the dedup option into the storage or not.